test/jdk/java/net/httpclient/CancelStreamedBodyTest.java

```diff
@@ -0,0 +1,432 @@
+        HttpTestHandler h1_chunkHandler = new HTTPSlowHandler();
+        InetSocketAddress sa = new InetSocketAddress(InetAddress.getLoopbackAddress(), 0);
+        httpTestServer = HttpTestServer.of(HttpServer.create(sa, 0));
+        httpTestServer.addHandler(h1_chunkHandler, "/http1/x/");
+        httpURI = "http://" + httpTestServer.serverAuthority() + "/http1/x/";
+
+        HttpsServer httpsServer = HttpsServer.create(sa, 0);
+        httpsServer.setHttpsConfigurator(new HttpsConfigurator(sslContext));
+        httpsTestServer = HttpTestServer.of(httpsServer);
+        httpsTestServer.addHandler(h1_chunkHandler, "/https1/x/");
+        httpsURI = "https://" + httpsTestServer.serverAuthority() + "/https1/x/";
+
+        // HTTP/2
+        HttpTestHandler h2_chunkedHandler = new HTTPSlowHandler();
+
+        http2TestServer = HttpTestServer.of(new Http2TestServer("localhost", false, 0));
+        http2TestServer.addHandler(h2_chunkedHandler, "/http2/x/");
+        http2URI = "http://" + http2TestServer.serverAuthority() + "/http2/x/";
+
+        https2TestServer = HttpTestServer.of(new Http2TestServer("localhost", true, sslContext));
+        https2TestServer.addHandler(h2_chunkedHandler, "/https2/x/");
+        https2URI = "https://" + https2TestServer.serverAuthority() + "/https2/x/";
+
+        serverCount.addAndGet(4);
+        httpTestServer.start();
+        httpsTestServer.start();
+        http2TestServer.start();
+        https2TestServer.start();
+    }
+
+    @AfterTest
+    public void teardown() throws Exception {
+        String sharedClientName =
+                sharedClient == null ? null : sharedClient.toString();
+        TRACKER.track(sharedClient);
+        sharedClient = null;
+        Thread.sleep(100);
+        AssertionError fail = TRACKER.check(500);
+        try {
+            httpTestServer.stop();
+            httpsTestServer.stop();
+            http2TestServer.stop();
+            https2TestServer.stop();
+        } finally {
+            if (fail != null) {
+                if (sharedClientName != null) {
+                    System.err.println("Shared client name is: " + sharedClientName);
+                }
+                throw fail;
+            }
+        }
+    }
+
+    /**
+     * A handler that slowly sends back a body to give time for the
+     * the request to get cancelled before the body is fully received.
+     */
+    static class HTTPSlowHandler implements HttpTestHandler {
+        @Override
+        public void handle(HttpTestExchange t) throws IOException {
+            try {
+                out.println("HTTPSlowHandler received request to " + t.getRequestURI());
+                System.err.println("HTTPSlowHandler received request to " + t.getRequestURI());
+
+                byte[] req;
+                try (InputStream is = t.getRequestBody()) {
+                    req = is.readAllBytes();
+                }
+                t.sendResponseHeaders(200, -1); // chunked/variable
```

Comment:
Unrelated to this PR - but this is extra confusing that our `HttpTestExchange` expects `-1` for chunked responses whereas the actual server `com.sun.net.httpserver.HttpExchange` expects `0` for chunked responses.

Reply:
> I'm not sure I want to change it ;-)

I wasn't expecting this to be changed :) At this point, it's now become a habit to pay extra attention to the value we pass for this method.